File: Python-projects/Data Structures and Algorithms/Programs/Week3_Assignment/remdup.py
def remdup(l):
    if len(l)<=1:
        return l
    else:
        nl=[]
        for i in range(0,len(l)):
            j=i
            for j in range(j,len(l)-1):
                if(l[i]==l[j+1]):
                    break;
            else:
                nl.append(l[i])        

        return nl        
    
a=[3,5,7,5,3,7,7,5,10];
b=[3,1,3,5];
c=[3,5,7,5,3,7,10];
d=[2,2,3,4,5,3,5]
e=[1,2,1]
print(remdup(d));                

File: Java/Recursion/RemDup.java
//Remove duplicates in a string

public class RemDup{
 public static boolean[] map = new boolean[26];
 public static void remdupStr(String str, int idx,String newStr){
    if(idx ==str.length()){
        System.out.println(newStr);
        return;
    }
    char currChar = str.charAt(idx);
    if(map[currChar-'a']){
        currChar++;
        remdupStr(str, idx+1, newStr);
    }
    else{
        map[currChar-'a'] = true;
        newStr+=currChar;
        remdupStr(str, idx+1, newStr);
    }
    

 }
    public static void main(String[] args) {
        remdupStr("axbsbabxxceds", 0, "");
    }

}